File: 169-Majority-Element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int, int> dic = new Dictionary<int, int>();
        for(int i = 0;i<nums.Length;i++){
            if(dic.ContainsKey(nums[i])){
                dic[nums[i]]++;
            }else{
              dic[nums[i]] = 1 ;  
            }
        }
        
        foreach (KeyValuePair<int, int> kvp in dic){
            if(kvp.Value > nums.Length/2){
                return kvp.Key;
            }
        }

        return -1 ;


    }
}

File: 169-Majority-Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let dic = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        if (dic.has(nums[i])) {
            dic.set(nums[i], dic.get(nums[i]) + 1);
        } else {
            dic.set(nums[i], 1);
        }
    }

    for (let [key, value] of dic) {
        if (value > nums.length / 2) {
            return key;
        }
    }

    return -1;
};
